tt_metal/impl/program/program.cpp

```diff
@@ -95,7 +95,7 @@ auto Program::semaphores_on_core(const CoreCoord &core) const {
 
 std::atomic<uint64_t> Program::program_counter = 0;
 
-Program::Program(): id(program_counter++),worker_crs_({}), local_circular_buffer_allocation_needed_(false) {
+Program::Program(): id(program_counter++),worker_crs_({}), local_circular_buffer_allocation_needed_(false), loaded(false) {
```

Comment:
can you add a comment explaining that loaded means it has been cached on device

Reply:
added comment in the header

Reply:
actually renamed to loaded_onto_device